types/node_jobs.go

```diff
@@ -20,58 +21,88 @@ type NodeJobType string
 
 const BLSToExecutionChangesNodeJobType NodeJobType = "BLS_TO_EXECUTION_CHANGES"
 const VoluntaryExitsNodeJobType NodeJobType = "VOLUNTARY_EXITS"
+const UnknownNodeJobType NodeJobType = "UNKNOWN"
 
 var NodeJobTypes = []NodeJobType{
 	BLSToExecutionChangesNodeJobType,
 	VoluntaryExitsNodeJobType,
 }
 
+func NewNodeJob(data []byte) (*NodeJob, error) {
+	j := &NodeJob{}
+	j.RawData = data
+	err := j.ParseData()
+	if err != nil {
+		return nil, err
+	}
+	return j, nil
+}
+
 type NodeJob struct {
 	ID                  string        `db:"id"`
 	CreatedTime         time.Time     `db:"created_time"`
 	SubmittedToNodeTime sql.NullTime  `db:"submitted_to_node_time"`
 	CompletedTime       sql.NullTime  `db:"completed_time"`
 	Type                NodeJobType   `db:"type"`
 	Status              NodeJobStatus `db:"status"`
-	Data                []byte        `db:"data"`
+	RawData             []byte        `db:"data"`
+	Data                interface{}   `db:"-"`
 }
 
-func (nj NodeJob) ToBLSToExecutionChangesNodeJob() (*BLSToExecutionChangesNodeJob, error) {
-	j := &BLSToExecutionChangesNodeJob{}
-	j.ID = nj.ID
-	j.CreatedTime = nj.CreatedTime
-	j.SubmittedToNodeTime = nj.SubmittedToNodeTime
-	j.CompletedTime = nj.CompletedTime
-	j.Type = nj.Type
-	j.Status = nj.Status
-	err := json.Unmarshal(nj.Data, &j.Data)
-	if err != nil {
-		return nil, err
+type NodeJobValidatorInfo struct {
+	ValidatorIndex      uint64 `db:"validatorindex"`
+	PublicKey           []byte `db:"pubkey"`
+	WithdrawCredentials []byte `db:"withdrawalcredentials"`
+	ExitEpoch           uint64 `db:"exitepoch"`
+	Status              string
+}
+
+// ParseData will try to unmarshal NodeJob.RawData into NodeJob.Data and determine NodeJob.Type by doing so. If it is not able to unmarshal any type it will return an error. It will sanitize NodeJob.RawData on success.
+func (nj *NodeJob) ParseData() error {
+	{
+		d := []*capella.SignedBLSToExecutionChange{}
+		err := json.Unmarshal(nj.RawData, &d)
+		if err == nil {
+			if nj.Type != "" && nj.Type != UnknownNodeJobType && nj.Type != BLSToExecutionChangesNodeJobType {
+				return fmt.Errorf("nodejob.RawData missmatches nodejob.Type (%v)", nj.Type)
+			}
+			nj.Type = BLSToExecutionChangesNodeJobType
+			nj.Data = d
+			return nj.SanitizeRawData()
+		}
 	}
-	return j, nil
+	{
+		//var d *VoluntaryExitsNodeJobData
+		var d *phase0.SignedVoluntaryExit
+		err := json.Unmarshal(nj.RawData, &d)
+		if err == nil && d.Message.Epoch != 0 {
+			if nj.Type != "" && nj.Type != UnknownNodeJobType && nj.Type != VoluntaryExitsNodeJobType {
+				return fmt.Errorf("nodejob.RawData missmatches nodejob.Type (%v)", nj.Type)
+			}
+			nj.Type = VoluntaryExitsNodeJobType
+			nj.Data = d
+			return nj.SanitizeRawData()
+		}
+	}
+	nj.Type = UnknownNodeJobType
+	return fmt.Errorf("can not unmarshal job-data")
```

Comment:
**Minor:** This should be changed to "cannot". Furthermore I suggest to capitalize the first letter as this is shown as flash messages on the webpage.